lib/test/Test/Pos/Core/AddressSpec.hs

```diff
@@ -6,22 +6,91 @@ module Test.Pos.Core.AddressSpec
-    prop "PK and HDW addresses with same public key are shown differently"
-         pkAndHdwAreShownDifferently
+spec = describe "Address" $ do
+    modifyMaxSuccess (min 10) $ do
+        prop "PK and HDW addresses with same public key are shown differently"
+            pkAndHdwAreShownDifferently
+
+    describe "Largest addresses" $ do
+        let genPubKeyAddrBoot = pure . makePubKeyAddressBoot . toPublic
+        largestAddressProp "PubKey address with BootstrapEra distribution"
+            genPubKeyAddrBoot largestPubKeyAddressBoot 43
+
+        let genPubKeyAddrSingleKey = pure . makePubKeyAddress
+                (IsBootstrapEraAddr False) . toPublic
+        largestAddressProp "PubKey address with SingleKey distribution"
+            genPubKeyAddrSingleKey largestPubKeyAddressSingleKey 78
+
+        let genHDAddrBoot :: SecretKey -> Gen Address
+            genHDAddrBoot sk = frequency
+                [ (1, genHDAddrBootEmptyPass sk)
+                , (5, genHDAddrBootSomePass sk)
+                ]
+
+            genHDAddrBoot' :: PassPhrase -> EncryptedSecretKey -> Word32 -> Word32 -> Address
+            genHDAddrBoot' passphrase esk accIdx addrIdx =
+                case deriveLvl2KeyPair
+                            (IsBootstrapEraAddr True)
+                            (ShouldCheckPassphrase False)
+                            passphrase
+                            esk
+                            accIdx
+                            addrIdx of
+                    Nothing        -> error "genHDAddrBoot' failed in tests"
+                    Just (addr, _) -> addr
+
+            genHDAddrBootEmptyPass sk =
+                genHDAddrBoot' emptyPassphrase (noPassEncrypt sk) <$>
+                arbitrary <*> arbitrary
+
+            genHDAddrBootSomePass (SecretKey sk) = do
+                passphrase <- arbitrary
+                esk <- mkEncSecret passphrase sk
+                genHDAddrBoot' passphrase esk <$> arbitrary <*> arbitrary
+
+        largestAddressProp "HD address with BootstrapEra distribution"
+            genHDAddrBoot largestHDAddressBoot 76
 
 pkAndHdwAreShownDifferently :: Bool -> PublicKey -> Bool
 pkAndHdwAreShownDifferently isBootstrap pk =
     show (makePubKeyAddress (IsBootstrapEraAddr isBootstrap) pk) /=
     (show @_ @Text (makePubKeyHdwAddress (IsBootstrapEraAddr isBootstrap)
                     (HDAddressPayload "pataq") pk))
+
+largestAddressProp :: Text -> (SecretKey -> Gen Address) -> Address -> Byte -> Spec
+largestAddressProp addressDescription genAddress largestAddress expectedLargestSize = do
+    describe (toString addressDescription) $ do
+
+        it (formatToString ("has the size we expect: "%memory) expectedLargestSize) $ do
+            biSize largestAddress `shouldBe` expectedLargestSize
+
+        prop "is indeed the largest address" $ do
+            forAll (deterministicKeyGen . BS.pack <$> vectorOf 32 arbitrary) $ \(_, sk) ->
```

Comment:
Isn't there an `Arbitrary` instance for `pk/sk`?

Reply:
There is, but it takes a key from a predefined pool of keys. That pool consists of only 128 keys. Here I want to consider more keys.